src/main/kotlin/domain/state/PlayerHit.kt

```diff
@@ -0,0 +1,20 @@
+package domain.state
+
+import domain.card.Card
+import domain.card.HandOfCards
+
+class PlayerHit(override val handOfCards: HandOfCards) : InProgress() {
+    override fun nextState(card: Card): State {
+        handOfCards.addCard(card)
+        if (handOfCards.isBust()) return Bust(handOfCards)
+        return this
+    }
+}
+
+class DealerHit(override val handOfCards: HandOfCards) : InProgress() {
+    override fun nextState(card: Card): State {
+        handOfCards.addCard(card)
+        if (handOfCards.isBust()) return Bust(handOfCards)
```

Comment:
```suggestion
        return if (handOfCards.isBust()) {
            Bust(handOfCards)
        } else 
            Stay(handOfCards)
        }
```
저는 개인적으로는 이런 형태의 구현을 더 선호하는데요, return 이 if 문 범위에 묶일 경우 논리 흐름을 다시 거슬러 올라가 생각할 필요가 적어지는 이점이 있습니다.

현재 구현하신 구조는, return 이 연속으로 선언되어 있어서 Stay 를 리턴하는 경우는 뭐지? 라고 다시 생각하게 되더라구요 ㅎㅎ

Reply:
return 조건문 형식으로 수정하였습니다!
저도 return을 lift하는 것이 더 가독성이 좋다고 생각됩니다.

다만 else를 지양하라는 객체지향 생활체조 원칙 때문에 이렇게 진행하였는데
해당 원칙은 자바를 기반으로 기술되어 코틀린에서는 예외가 있을 수 있다고 생각됩니다!